react/the_react_ecosystem/fetching-data.md

```diff
@@ -0,0 +1,297 @@
+### Introduction
+
+Up to this point we have been using React to build client side applications with interactive user interfaces, but what if we want fetch data from the internet? In order to create full fledged web applications, we need some way to get data from external sources and dynamically displaying it on the client.
+
+In this lesson, we'll explore the ins and outs of fetching data in React, starting with the basics of making API calls, managing component state, and handling asynchronous operations using JavaScript's ~fetch~ function.
```

Comment:
Since the learners would've already done data fetching in the memory card and shopping cart projects, it's good to mention it here.

```suggestion
In this lesson, we'll explore the ins and outs of fetching data in React, starting with the basics of making API calls, managing component state, and handling asynchronous operations using JavaScript's `fetch` function. You've already performed data fetching in earlier projects, so some stuff covered in the lesson will be familiar to you. Revision doesn't hurt!
```